misk/src/main/kotlin/misk/web/interceptors/RequestLoggingInterceptor.kt

```diff
@@ -23,46 +25,55 @@ private val logger = getLogger<RequestLoggingInterceptor>()
  */
 class RequestLoggingInterceptor internal constructor(
   private val action: Action,
-  private val sampling: Double,
-  private val includeBody: Boolean,
   private val caller: ActionScoped<MiskCaller?>,
   private val ticker: Ticker,
   private val random: ThreadLocalRandom,
+  private val logRateLimiter: LogRateLimiter,
+  private val bodySampling: Double,
+  private val errorBodySampling: Double,
   private val bodyCapture: RequestResponseCapture
 ) : NetworkInterceptor {
   @Singleton
   class Factory @Inject internal constructor(
     private val caller: @JvmSuppressWildcards ActionScoped<MiskCaller?>,
     private val ticker: Ticker,
     private val random: ThreadLocalRandom,
-    private val bodyCapture: RequestResponseCapture
+    private val bodyCapture: RequestResponseCapture,
+    private val logRateLimiter: LogRateLimiter
   ) : NetworkInterceptor.Factory {
     override fun create(action: Action): NetworkInterceptor? {
       val logRequestResponse = action.function.findAnnotation<LogRequestResponse>() ?: return null
-      require(0.0 < logRequestResponse.sampling && logRequestResponse.sampling <= 1.0) {
+      require(logRequestResponse.rateLimiting >= 0L) {
+        "${action.name} @LogRequestResponse rateLimiting must be >= 0"
+      }
+      require(logRequestResponse.errorRateLimiting >= 0L) {
+        "${action.name} @LogRequestResponse errorRateLimiting must be >= 0"
+      }
+      require(0.0 <= logRequestResponse.bodySampling && logRequestResponse.bodySampling <= 1.0) {
+        "${action.name} @LogRequestResponse sampling must be in the range (0.0, 1.0]"
+      }
+      require(
+        0.0 <= logRequestResponse.errorBodySampling && logRequestResponse.errorBodySampling <= 1.0) {
         "${action.name} @LogRequestResponse sampling must be in the range (0.0, 1.0]"
       }
 
+      logRateLimiter.addRateLimits(action.name, logRequestResponse.rateLimiting,
+        logRequestResponse.errorRateLimiting)
       return RequestLoggingInterceptor(
         action,
-        logRequestResponse.sampling,
-        logRequestResponse.includeBody,
         caller,
         ticker,
         random,
+        logRateLimiter,
+        logRequestResponse.bodySampling,
+        logRequestResponse.errorBodySampling,
```

Comment:
Good point I didn't need to pass it in when I had `RealLogRateLimiter.addRateLimits`, but because I'm updating `tryAcquire` by passing in `ratePerSecond` looks like I need to pass them into the interceptor now